xbmc/addons/AddonManager.cpp

```diff
@@ -608,8 +610,15 @@ bool CAddonMgr::FindAddon(const std::string& addonId, const AddonVersion& addonV
   CLog::Log(LOGINFO, "CAddonMgr::{}: {} v{} installed", __FUNCTION__, addonId,
             addonVersion.asString());
 
+  bool isNewInstall = !IsAddonInstalled(addonId);
+
   m_installedAddons[addonId] = it->second; // insert/replace entry
 
+  if (isNewInstall)
```

Comment:
As you do not need the variable elsewhere, why not just 'if (!IsAddonInstalled(addonId))'?

Reply:
can do this, but would then need to move `m_installedAddons[addonId] = it->second; // insert/replace entry` into the if (which looks much better, i agree)

Reply:
Not sure I get that. Why must m_installed ... be within the if clause?

Reply:
because when `m_installed...` inserts, it potentially changes the result of `IsAddonInstalled(`...
behind the `if` is correct here, not within the `if`

Reply:
okay. makes sense.